### ISSUE TITLE
vmstorage panic when removal queue full

### ISSUE THREAD

**Issue by faceair** [VictoriaMetrics/VictoriaMetrics#1313]
**Describe the bug**
I am using vm on NFS, it seems to be related to this issue  https://github.com/VictoriaMetrics/VictoriaMetrics/issues/1142
```
2021-05-21T10:34:22.763Z	panic	VictoriaMetrics/lib/fs/dir_remover.go:37	FATAL: cannot schedule /data/alibjg-common-victoriametrics5-vm/vmstorage/indexdb/1678213C00BBB381/4353_11_168105D87787AFCA for removal, since the removal queue is full (1024 entries)
panic: FATAL: cannot schedule /data/alibjg-common-victoriametrics5-vm/vmstorage/indexdb/1678213C00BBB381/4353_11_168105D87787AFCA for removal, since the removal queue is full (1024 entries)

goroutine 356 [running]:
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logMessage(0x9b86ca, 0x5, 0xc0dc3a49c0, 0xb6, 0x4)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:270 +0xc69
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevelSkipframes(0x1, 0x9b86ca, 0x5, 0x9d7b5b, 0x53, 0xc424da3838, 0x2, 0x2)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:138 +0xd1
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevel(...)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:130
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.Panicf(...)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:126
github.com/VictoriaMetrics/VictoriaMetrics/lib/fs.mustRemoveAll(0xc5eb704d90, 0x63, 0xc23c14c6d0, 0x63)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/fs/dir_remover.go:37 +0x26a
github.com/VictoriaMetrics/VictoriaMetrics/lib/fs.MustRemoveAllWithDoneCallback(...)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/fs/fs.go:213
github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset.runTransaction(0xc032538088, 0xc00c834500, 0x4a, 0xc511238840, 0x5f, 0x0, 0x0)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset/table.go:1271 +0x372
github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset.(*Table).mergeParts(0xc032538000, 0xc267830840, 0x8, 0x8, 0xc00af0b2c0, 0xc267830800, 0x0, 0x0)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset/table.go:925 +0xe2a
github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset.(*Table).mergeExistingParts(0xc032538000, 0x3b9aca00, 0x0, 0x0)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset/table.go:766 +0xed
github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset.(*Table).partMerger(0xc032538000, 0x0, 0x0)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset/table.go:780 +0x7a
github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset.(*Table).startPartMergers.func1(0xc032538000)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset/table.go:748 +0x32
created by github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset.(*Table).startPartMergers
	github.com/VictoriaMetrics/VictoriaMetrics/lib/mergeset/table.go:747 +0x69
```
I checked that the number of empty folders is less than 100 in most cases.
This time the problem occurred when we tweaked the vmagent job to add about a million write datapoints.

**Screenshots**
![image](https://user-images.githubusercontent.com/3659110/119129153-1f09cb00-ba69-11eb-8641-104b1fbfcbac.png)

**Version**
vmstorage-20210501-072056-tags-v1.59.0-cluster-0-g9aa44a276

**Comment by valyala** [comment]
@faceair , could you verify whether 8e2985b53d314ca387e8ce3692cd918ca79f752a fixes the issue?

**Comment by valyala** [comment]
Oops, it looks like the fix doesn't work as expected. New fix is available in the commit 89e1a45cdb35ecdcacaacffb0a45a23784d72281 .

**Comment by faceair** [comment]
![image](https://user-images.githubusercontent.com/3659110/119165259-e3362c00-ba8f-11eb-887a-950fd26201fd.png)

![image](https://user-images.githubusercontent.com/3659110/119165287-eaf5d080-ba8f-11eb-80e8-952f33a7f0f1.png)

![image](https://user-images.githubusercontent.com/3659110/119165319-f21cde80-ba8f-11eb-8c48-60c2004cb2f8.png)

pprof [goroutine.txt](https://github.com/VictoriaMetrics/VictoriaMetrics/files/6523506/goroutine.txt)

When I use the latest commit, the data write is blocked.

**Comment by valyala** [comment]
@faceair , I received the `When I use the latest commit, the data write is blocked` message in my email, but I don't see it here. Is it still valid? If yes, could you post it here again?

**Comment by faceair** [comment]
I ran into some performance issues with NFS itself last Friday after deploying the latest commits, which have since recovered automatically.
Now the latest commit is working fine and all seems ok. I may need to do some additional monitoring of my own NFS to troubleshoot some of its own issues.

**Comment by faceair** [comment]
![image](https://user-images.githubusercontent.com/3659110/119292024-20faa680-bc82-11eb-926a-b21aa2a822b2.png)
I confirmed that this issue is caused by my NFS write latency increase, I will upgrade my NFS type and thank you for the fix.

**Comment by valyala** [comment]
Thanks for the update, @faceair !

The bugfix has been included in [v1.60.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue then. Please re-open it if `vmstorage` still panics with `the removal queue is full` message after cluster upgrade to v1.60.0 .
